apps/testing-harness/src/app/child.component.spec.ts

```diff
@@ -1,38 +1,84 @@
 import { render } from '@testing-library/angular';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatCheckboxHarness } from '@angular/material/checkbox/testing';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { MatSliderHarness } from '@angular/material/slider/testing';
+import { HarnessLoader } from '@angular/cdk/testing';
 import { ChildComponent } from './child.component';
 
 describe('ChildComponent', () => {
+  let component: ChildComponent;
+  let loader: HarnessLoader;
+  beforeEach(async () => {
+    const { fixture } = await render(ChildComponent);
+    component = fixture.componentInstance;
+    loader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
   describe('When init', () => {
     test('Then show 1 slider, 3 checkboxes, 4 inputs, 2 buttons', async () => {
-      await render(ChildComponent);
+      const [inputs, checkboxes, sliders, buttons] = await Promise.all([
+        loader.getAllHarnesses(MatInputHarness),
+        loader.getAllHarnesses(MatCheckboxHarness),
+        loader.getAllHarnesses(MatSliderHarness),
+        loader.getAllHarnesses(MatButtonHarness),
+      ]);
+
+      expect(inputs.length).toBe(4);
+      expect(checkboxes.length).toBe(3);
+      expect(sliders.length).toBe(1);
+      expect(buttons.length).toBe(2);
     });
 
     test('Then initial value of slider thumb is 0', async () => {
-      await render(ChildComponent);
+      const slider = await loader.getHarness(MatSliderHarness);
+      const endThumb = await slider.getEndThumb();
+      const value = await endThumb.getValue();
+      expect(value).toBe(0);
     });
   });
 
   describe('Given maxValue set to 109', () => {
     test('Then slider max value is 109', async () => {
-      await render(ChildComponent);
+      const slider = await loader.getHarness(MatSliderHarness);
+      component.max = 109;
```

Comment:
The thing I don't like is that you use implementation detail. I always avoid using it and I create my test from the user point of view and I cannot use anything else that the user is not seeing. This way, if the implementation change, my test should always pass.